src/cmap/test/event.rs

```diff
@@ -110,6 +114,20 @@ impl EventSubscriber<'_> {
             .ok()
             .flatten()
     }
+
+    /// Returns the received events without waiting for any more.
+    pub fn all<F>(&mut self, filter: F) -> Vec<Event>
+    where
+        F: Fn(&Event) -> bool,
+    {
+        let mut events = Vec::new();
```

Comment:
So this invocation would wait for the entire stream to be consumed and then collect all the results I think (`collect` is `async`), whereas we just want to aggregate any events that are buffered in the stream already without waiting for any more.

Reply:
Ah, looks like you're right. That's a little disappointing; this feels like the exact type of thing we should be able to do with the Stream API instead of having to do it manually.